olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -203,6 +198,7 @@ client::CancellationToken VersionedLayerClientImpl::PrefetchTiles(
   auto layer_id = layer_id_;
   auto settings = settings_;
   auto pending_requests = pending_requests_;
+  auto version = catalog_version_.load();
```

Comment:
This is incorrect, we get the version inside the task: 
`GetVersion(request.GetBillingTag(), OnlineIfNotFound, context);`